### PR TITLE
cv uploads to firebase

### LINKED ISSUE #120: Add CV to user data to make it uploadable to Firebase.

### PR DESCRIPTION
**Pull request by lpatel99**

### CODE REVIEW

**Comment by arnauds5** [comment]
There is an empty FakeUserDataDatabase (in androidTest/java/com/sdp13epfl2021/projmag/database/fake/) that need to have your new function, you can just put this
```
    override fun pushCv(
        cv: CurriculumVitae,
        onSuccess: () -> Unit,
        onFailure: (java.lang.Exception) -> Unit
    ) {
        TODO("Not yet implemented")
    }
```

File: app/src/main/java/com/sdp13epfl2021/projmag/activities/CVCreationActivity.kt [hunk 1]
```diff
@@ -35,7 +37,9 @@ class CVCreationActivity : AppCompatActivity() {
 
     private fun onFinish() {
         val cv = buildCV()
-        /* DO SOMETHING WITH IT */
+        val utils = Utils.getInstance(this)
+        val userDataFirebase = utils.userDataDatabase
+        userDataFirebase.pushCv(cv, { Toast.makeText(this, "Success!", Toast.LENGTH_LONG)}, {Toast.makeText(this, "Failure!", Toast.LENGTH_LONG)})
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/activities/CVCreationActivity.kt]: The Toast are now shown, and maybe using resources strings would be better
```suggestion
        userDataFirebase.pushCv(cv, {
            Toast.makeText(this, "Success!", Toast.LENGTH_LONG).show()
        }, {
            Toast.makeText(this, "Failure!", Toast.LENGTH_LONG).show()
        })
```

File: app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataDatabase.kt [hunk 2]
```diff
@@ -65,4 +66,10 @@ interface UserDataDatabase {
         onSuccess: () -> Unit,
         onFailure: (Exception) -> Unit
     )
+
+    fun pushCv(
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataDatabase.kt:79]: The interface need to have some (basic) documentation,

File: app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt [hunk 3]
```diff
@@ -108,4 +109,19 @@ class UserDataFirebase(
...
+
+    override fun pushCv(
+        cv: CurriculumVitae,
+        onSuccess: () -> Unit,
+        onFailure: (java.lang.Exception) -> Unit
+    ) {
+        getUserDoc()?.let {
+            doc ->
+            doc.set(
+                hashMapOf("cv" to cv),
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt]: Using string literval inside the code could be avoided by using constants in the companion object,
where we can put something like that :
```
/**
 *  The field containing cv
 */
const val CV_FIELD = "cv"
```

File: app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt [hunk 4]
```diff
@@ -108,4 +109,19 @@ class UserDataFirebase(
             )
         }
     }
+
+    override fun pushCv(
+        cv: CurriculumVitae,
+        onSuccess: () -> Unit,
+        onFailure: (java.lang.Exception) -> Unit
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt]: Exception is a little more readable than java.lang.Exception
```suggestion
        onFailure: (Exception) -> Unit
```

File: app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt [hunk 5]
```diff
@@ -108,4 +109,19 @@ class UserDataFirebase(
...
+    ) {
+        getUserDoc()?.let {
+            doc ->
+            doc.set(
+                hashMapOf("cv" to cv),
+                SetOptions.merge()
+            ).addOnSuccessListener { onSuccess }
+                .addOnFailureListener(onFailure)
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt]: The onSuccess was never call with ()
```suggestion
            ).addOnSuccessListener { onSuccess() }
                .addOnFailureListener(onFailure)
```

File: app/src/main/java/com/sdp13epfl2021/projmag/MainActivity.kt [hunk 6]
```diff
@@ -9,6 +9,7 @@ import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import com.sdp13epfl2021.projmag.activities.CVCreationActivity
```
Comment (asfoury) [inline app/src/main/java/com/sdp13epfl2021/projmag/MainActivity.kt:12]: Is this import used in the MainActivity?

**Review by asfoury — APPROVED** [review]

**Review by arnauds5 — APPROVED** [review]
Normaly this need to have tests (with mockito), but it seems ok for now
(When someone will add the getCv/etc in next week, we will need to write tests)

**Pull request merged**